src/sys/unix.rs

```diff
@@ -691,7 +691,7 @@ impl SockAddr {
     pub fn is_unnamed(&self) -> bool {
         self.as_sockaddr_un()
             .map(|storage| {
-                self.len() == offset_of_path(storage) as u32
+                self.len() == offset_of_path(storage).try_into().unwrap()
```

Comment:
Please use `as _` instead. I don't want to introduce a panic branch.

Reply:
Same for the other changes.

Reply:
but, if `as _` fails, wont it make even harder to debug if type conversion is somehow messed up ?

Reply:
`offset_of_path` max is 108 bytes (off the top of my head), so it won't ever fail.

Reply:
then it would not panic too, right ? why not try_into() ? This i guess is a much cleaner way, other way being conditional compilation using `#[cfg(...)]`

Reply:
> then it would not panic too, right ?

No, why would it panic?

> why not try_into() ?

As I said above that introduces a panic code branch, something we don't want.

> This i guess is a much cleaner way, other way being conditional compilation using `#[cfg(...)]`

Just use `as _`. The existing code was fine, Android just uses a different type `i32` instead of `u32`, both can easily fit the number 108.

Reply:
I still don't get it (maybe nub at rust).
Please do push the required changes onto this PR

Reply:
Wow, didn't know that we can use `_` with `as`. Thought you were suggesting me to fill-in-the-blank after `as u32/i32`. Thanks learned something new today.

Don't want to sound argumentative but isn't it safer to use try_into() when we have no control over upstream crate (here `libc` which provides with `socklen_t`). Even if its a policy, why no panic blocks, is it just code optimization or there is something else deeper here ?

Reply:
Understood. Thanks for your patience 🙏